ldm/modules/embedding_manager.py

```diff
@@ -302,6 +292,92 @@ def add_embedding(self, token_str, embedding, full):
         self.string_to_token_dict[token_str] = token
         self.string_to_param_dict[token_str] = torch.nn.Parameter(embedding)
 
+    def parse_embedding(self, embedding_file: str):
+        file_type = embedding_file.split('.')[-1]
+        if file_type == 'pt':
+            return self.parse_embedding_pt(embedding_file)
+        elif file_type == 'bin':
+            return self.parse_embedding_bin(embedding_file)
+        else:
+            print(f'>> Not a recognized embedding file: {embedding_file}')
+
+    def parse_embedding_pt(self, embedding_file):
+        embedding_ckpt = torch.load(embedding_file, map_location='cpu')
+        embedding_info = {}
+
+        # Check if valid embedding file
+        if 'string_to_token' and 'string_to_param' in embedding_ckpt:
+
+            # Catch variants that do not have the expected keys or values.
+            try:
```

Comment:
I have put a try: around this section in order to catch and fix two different variants found in the wild, including the one reported in Issue #1829 (a 12-vector version!). I tested and all seems good, but please have a look.

Reply:
Seems good in code. I'll give it a run soon and see if it holds up.